src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -126,8 +126,9 @@ trait SyntheticMethods extends ast.TreeDSL {
       createSwitchMethod(name, accessors.indices, returnType)(idx => caseFn(accessors(idx)))
 
     def productElementNameMethod = {
-      val constrParamAccessors = clazz.constrParamAccessors
-      createSwitchMethod(nme.productElementName, constrParamAccessors.indices, StringTpe)(idx => LIT(constrParamAccessors(idx).name.dropLocal.decode))
+      createSwitchMethod(nme.productElementName, accessors.indices, StringTpe)(idx =>
+        LIT(accessors(idx).name.dropLocal.decode)
+      )
```

Comment:
That was bad advice because mangled names. `clazz.constrParamAccessors.take(arity)`.

You might think these go through `paramLists` aka `paramss` but [actually no](https://github.com/scala/scala/blob/2.13.x/src/reflect/scala/reflect/internal/Symbols.scala#L2137).

https://github.com/scala/scala/pull/9645/files#diff-205537ac4c08ea690ada72e398df0018dcaf2a7c4987c0d8d8df322314469578R129